initial_settings/initial.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/python -tt
+
+import re
+import subprocess
+
+
+class DnfUpgrageTo():
+    def create_repo(self):
+        subprocess.check_call(['echo -ne "[dnf-pull-requests]\nname=dnf-pull-requests\nbaseurl=https://copr-be.'
+                               'cloud.fedoraproject.org/results/rpmsoftwaremanagement/dnf-pull-requests/'
+                               'fedora-rawhide-x86_64/\nenabled=1\ngpgcheck=0" > /etc/yum.repos.d/'
+                               'dnf-pull-requests.repo'], shell=True)
+
+    def dnf_version(self):
+        f = open("/initial_settings/ci-dnf-stack.log")
+        version = []
+        for line in f:
+            m = re.search("(dnf-\d+[.]\d+[.]\d+-\d+[.]git[.][a-zA-Z0-9.]+)[.]fc", line)
+            if m:
+                version.append(m.group(1))
+        version = list(set(version))
```

Comment:
I think I meant why do you convert it to the list? you can still use `len` and `pop`.